Add section on using Cedar as a Library to README (#468)
cedar-policy/cedar: README.md

Which expresses that 2.4.2 is the minimum version of Cedar you accept, and you implicitly accept anything newer that is semver-compatible. See <https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html>.

> ## Using Cedar as a Library

Reviewer: ```suggestion
## Limits in Cedar
```

> Cedar is built to be a general-purpose library. As such, we can't anticipate every use case. We don't provide limits on e.g., policy length, schema length, depth of nested expressions, complexity of `like` operator expressions or set operations. If any of these variables can be changed by untrusted actors in your application, you almost certainly want to choose and enforce some limits on these variables.

Reviewer: ```suggestion
Cedar is built to be a general-purpose library. The Cedar library itself doesn't provide limits on e.g., policy length, schema length, `EntityId` length, depth of nested expressions, complexity of `like` expressions, or set operations. If any of these can be controlled by untrusted actors in your application, you almost certainly want to choose and enforce some limits on these variables, appropriate to your application.
```
